### PR TITLE
Tgp 1370: storing latest on get

### PR DESCRIPTION
**Pull request by besscerule**

### CODE REVIEW

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/models/response/Pagination.scala [hunk 1]
```diff
@@ -41,10 +41,13 @@ object Pagination {
     val page                 = pageOpt.getOrElse(localStartingPage)
     val mod                  = totalRecords % size
     val totalRecordsMinusMod = totalRecords - mod
-    val totalPages           = ((totalRecordsMinusMod / size) + 1).toInt
+    val totalPages           = (if (mod == 0 && totalRecords != 0) {
+                        totalRecordsMinusMod / size
+                      } else {
+                        (totalRecordsMinusMod / size) + 1
+                      }).toInt
```
Comment (yohan-oc) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/models/response/Pagination.scala:48]: Indentation looks nice this way :D 
`val totalPages           = {
  if (mod == 0 && totalRecords != 0) {
    totalRecordsMinusMod / size
  } else {
    (totalRecordsMinusMod / size) + 1
  }
}.toInt`

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala [hunk 2]
```diff
@@ -45,14 +42,16 @@ class RecordsRepository @Inject() (
       )
     ) {
 
-  private def byEori(eori: String): Bson = Filters.equal("eori", eori)
+  private def byEori(eori: String): Bson          = Filters.equal("eori", eori)
+  private def byEoriAndActive(eori: String): Bson =
```
Comment (yohan-oc) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala:46]: Shall we add couple of unit test to cover this. We already have inactiveGoodsItemRecords in the spec which is not being used.

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/models/response/Pagination.scala (hunk 1 above)
Comment (yohan-oc) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/models/response/Pagination.scala:48]: Indentation looks nice this way :D
```
val totalPages           = { 
    if (mod == 0 && totalRecords != 0) { 
       totalRecordsMinusMod / size 
    } else { 
       (totalRecordsMinusMod / size) + 1 
    } 
}.toInt
```

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala (hunk 2 above)
Comment (yohan-oc) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala:46]: Shall we add couple of unit test to cover this. We already have `inactiveGoodsItemRecords` in the spec which is not being used.

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala [hunk 3]
```diff
@@ -106,30 +105,6 @@ class RecordsRepository @Inject() (
...
-  def insert(record: GoodsItemRecord): Future[Done] =
-    collection
-      .insertOne(
-        record
-      )
-      .toFuture()
-      .map(_ => Done)
-
   def saveRecord(eori: String, record: GoodsItemRecord): Future[Boolean] =
     collection
```
Comment (yohan-oc) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala:109]: `saveRecord` is no longer in use

**Review by yohan-oc — APPROVED** [review]
happy with the changes :)

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/models/response/Pagination.scala (hunk 1 above)
Comment (besscerule) [reply app/uk/gov/hmrc/tradergoodsprofilesdatastore/models/response/Pagination.scala]: done

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala (hunk 3 above)
Comment (besscerule) [reply app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala]: done

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala (hunk 2 above)
Comment (besscerule) [reply app/uk/gov/hmrc/tradergoodsprofilesdatastore/repositories/RecordsRepository.scala:46]: done

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/CreateRecordController.scala [hunk 4]
```diff
@@ -16,33 +16,33 @@
...
+    with Logging {
 
-  def create(eori: String): Action[CreateRecordRequest] =
+  def createRecord(eori: String): Action[CreateRecordRequest] =
     identify.async(parse.json[CreateRecordRequest]) { implicit request =>
-      routerConnector.createRecord(request.body).flatMap { goodsItemRecord =>
-        recordsRepository
-          .insert(goodsItemRecord)
-          .map(_ => Ok(Json.toJson(CreateRecordResponse(goodsItemRecord.recordId))))
+      routerConnector.createRecord(request.body, eori).map(recordId => Ok(recordId)) transform {
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/CreateRecordController.scala:41]: use recover if you only want to handle failures

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DeleteRecordController.scala [hunk 5]
```diff
@@ -16,35 +16,32 @@
...
-      case Some(goodsItemRecords) =>
-        routerConnector.deleteRecord(eori, recordId, goodsItemRecords.actorId).flatMap { case Done =>
-          recordsRepository.delete(eori, recordId).map {
-            case true  => NoContent
-            case false => InternalServerError("Failed to delete the record from db")
-          }
-        }
-      case None                   =>
-        Future.successful(NotFound)
+    routerConnector.deleteRecord(eori, recordId).map(response => response) transform {
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/DeleteRecordController.scala:39]: Sounds like it should be .map(_ => NoContent).recover...{}

File: app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/GetRecordsController.scala [hunk 6]
```diff
@@ -65,5 +71,13 @@ class GetRecordsController @Inject() (
     routerConnector.getRecords(eori, page = Some(recursiveStartingPage), size = Some(1)).map { recordsResponse =>
       Ok(Json.toJson(recordsResponse.pagination.totalRecords))
     }
+    routerConnector
+      .getRecords(eori, page = Some(recursiveStartingPage), size = Some(1))
+      .map(recordsResponse => Ok(Json.toJson(recordsResponse.pagination.totalRecords))) transform {
```
Comment (liandra-giacomine) [inline app/uk/gov/hmrc/tradergoodsprofilesdatastore/controllers/GetRecordsController.scala:76]: recover

**Pull request merged**